packages/urql/updates/index.js

```diff
@@ -93,31 +132,44 @@ export default {
   },
   Subscription: {
     comments: (result, args, cache, info) => {
-      console.log('!!!!! comments - result, args, info:', result, args, info)
+      makeAppendToPaginatedSetResolver({
+        parentType: 'Post',
+        parentId: result?.comments?.post?.id,
+        fieldName: 'comments'
+      })(result, args, cache, info)
     },
+
     updates: (result, args, cache, info) => {
       const update = result?.updates
 
       switch (update?.__typename) {
         case 'Message': {
-          console.log('!!!! updates TODO: new Message. Increment Messages tab badge', result, args)
-          // TODO: URQL - This one may want an updater to keep from killing the cache with each message
-          // leaving the cache invalidation off for now.
-          // cache.invalidate({ __typename: 'MessageThread', id: update?.messageThread?.id })
-          break
+          makeAppendToPaginatedSetResolver({
+            parentType: 'MessageThread',
+            parentId: update?.messageThread?.id,
+            fieldName: 'messages'
+          })(result, args, cache, info)
+          return
         }
+
         case 'MessageThread': {
-          console.log('!!!! updates TODO: new MessageThread. Increment Messages tab badge', result, args)
-          cache.invalidate({ __typename: 'MessageThread', id: update?.messageThread?.id })
-          break
+          cache.updateQuery({ query: meQuery }, ({ me }) => {
+            if (!me) return null
+            return { me: { ...me, unseenThreadCount: me.unseenThreadCount + 1 } }
+          })
+          return
         }
+
         case 'Notification': {
-          console.log('!!!! updates TODO: new Notification. Increment Notifications badge', result, args)
-          cache.invalidate('Query', 'notifications')
-          break
+          cache.updateQuery({ query: meQuery }, ({ me }) => {
+            if (!me) return null
+            return { me: { ...me, newNotificationCount: me.newNotificationCount + 1 } }
+          })
+          return
         }
+
         default: {
-          console.log('!!! Unhandled update from updates subscription', result)
+          console.log('ℹ️ Unhandled update from updates subscription', result, args)
```

Comment:
what do you think: should we use a console.warn for something like this?

Reply:
No, I don't think it needs anything more than this. Nothing bad really happened if we hit this, and we'd likely only see it if we were debugging and in the process of adding a new updates type.